src/Microsoft.DotNet.ArcadeAzureIntegration/DefaultIdentityTokenCredential.cs

```diff
@@ -0,0 +1,137 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if !NET472_OR_GREATER
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Azure.Core;
+using Azure.Identity;
+
+namespace Microsoft.DotNet.ArcadeAzureIntegration;
+
+
+// This implementation of TokenCredential will try cover all common ways to
+// authenticate to Azure services used in Arcade tooling
+public class DefaultIdentityTokenCredential : ChainedTokenCredential
+{
+    public DefaultIdentityTokenCredential()
+        : this(new DefaultIdentityTokenCredentialOptions())
+    {
+    }
+
+    public DefaultIdentityTokenCredential(DefaultIdentityTokenCredentialOptions options)
+        : base(CreateAvailableTokenCredentials(options))
+    {
+    }
+
+    private static TokenCredential[] CreateAvailableTokenCredentials(DefaultIdentityTokenCredentialOptions options)
+    {
+        var tokenCredentials = new List<TokenCredential>();
```

Comment:
Think this is a bit cleaner, not sure what you prefer
```suggestion
        List<TokenCredential> tokenCredentials = [];
```